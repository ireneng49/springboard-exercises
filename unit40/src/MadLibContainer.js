import React, { useState } from "react";
import MadLibForm from "./MadLibForm";
import MadLib from "./MadLib";
import './Container.css'

const MadLibContainer = () => {
    const [displayLib, setDisplayLib] = useState(false);
    const [madlibValues, setMadlibValues] = useState(null);

    return (
        <div className="Container">
            <h1>Mad Libs!</h1>
            {displayLib ? <MadLib
                madlibValues={madlibValues}
                setMadlibValues={setMadlibValues}
                setDisplayLib={setDisplayLib}/>
                : <MadLibForm
                    setDisplayLib={setDisplayLib}
                    setMadlibValues={setMadlibValues}/>}


        </div>
    );
}

export default MadLibContainer;