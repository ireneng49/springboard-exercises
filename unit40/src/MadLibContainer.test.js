import React from "react";
import { render } from "@testing-library/react";
import MadLibContainer from './MadLibContainer'

it("renders without crashing", function () {
  render(<MadLibContainer />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<MadLibContainer />);
  expect(asFragment()).toMatchSnapshot();
});

// it("Should return to creation form upon clicking reset button", function () {
//     const madlib = render(<MadLib madlibValues={madlibValues}
//         setMadlibValues={setMadlibValues}
//         setDisplayLib={setDisplayLib} />);

//     expect(madlib.getByText('There was a pink fish who loved a scaly dog.')).toBeInTheDocument();

//     const resetButton = madlib.getByText('Make Another Story')
//     expect(resetButton).toBeInTheDocument()
//     fireEvent.click(resetButton)
//     // expect(madlib.queryByText('There was a pink fish who loved a scaly dog.')).not.toBeVisible();
// })