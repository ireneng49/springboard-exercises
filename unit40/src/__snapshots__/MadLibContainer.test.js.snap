// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Container"
  >
    <h1>
      Mad Libs!
    </h1>
    <form
      style="text-align: center;"
    >
      <label
        for="noun"
        style="display: block; margin-top: 5px;"
      >
        Noun
      </label>
      <input
        id="noun"
        name="noun"
        placeholder="noun"
        type="text"
        value=""
      />
      <label
        for="noun2"
        style="display: block; margin-top: 5px;"
      >
        Noun 2
      </label>
      <input
        id="noun2"
        name="noun2"
        placeholder="noun 2"
        type="text"
        value=""
      />
      <label
        for="adjective"
        style="display: block; margin-top: 5px;"
      >
        Adjective
      </label>
      <input
        id="adjective"
        name="adjective"
        placeholder="adjective"
        type="text"
        value=""
      />
      <label
        for="color"
        style="display: block; margin-top: 5px;"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
      <label
        for="story"
        style="display: block; margin-top: 20px;"
      >
        Choose a Story
      </label>
      <select
        id="story"
        name="story"
      >
        <option
          value="Once Upon a Time"
        >
          Once Upon a Time
        </option>
        <option
          value="True Love"
        >
          True Love
        </option>
        <option
          value="Silly Song"
        >
          Silly Song
        </option>
        <option
          value="Neighborhood"
        >
          Neighborhood
        </option>
        <option
          value="Nature Documentary"
        >
          Nature Documentary
        </option>
        <option
          value="Grocery Shopping"
        >
          Grocery Shopping
        </option>
        <option
          value="Car Repair"
        >
          Car Repair
        </option>
        <option
          value="Pet Store"
        >
          Pet Store
        </option>
      </select>
      <button
        style="background-color: lightgreen; border-radius: 5px; padding: 5px 10px; font-weight: 700; border: 2px solid black; display: block; margin-top: 10px; margin-left: 40px;"
        type="submit"
      >
        Create Story
      </button>
    </form>
  </div>
</DocumentFragment>
`;
